kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/Kms.java

```diff
@@ -14,26 +14,25 @@
 
 /**
  * Abstracts the KMS operations needed for Envelope Encryption
- * @param <K> The type of Key Encryption Key id.
  * @param <E> The type of encrypted Data Encryption Key.
  */
-public interface Kms<K, E> {
+public interface Kms<E> {
 
     /**
      * Asynchronously generates a new Data Encryption Key (DEK) and returns it together with the same DEK wrapped by the Key Encryption Key (KEK) given
      * by the {@code kekRef},
-     * The returned encrypted DEK can later be decrypted with {@link Kms#decryptEdek(Object, Object)}.
+     * The returned encrypted DEK can later be decrypted with {@link Kms#decryptEdek(KekId, Object)}.
      * @param kekRef The key encryption key used to encrypt the generated data encryption key.
      * @return A completion stage for the wrapped data encryption key.
      * @throws UnknownKeyException If the kek was not known to this KMS.
      * @throws InvalidKeyUsageException If the given kek was not intended for key wrapping.
      * @throws KmsException For other exceptions.
      */
     @NonNull
-    CompletionStage<DekPair<E>> generateDekPair(@NonNull K kekRef);
+    CompletionStage<DekPair<E>> generateDekPair(@NonNull KekId kekRef);
```

Comment:
We should probably do 
```suggestion
    CompletionStage<DekPair<E>> generateDekPair(@NonNull KekId kekId);
```

However even if we don't proceed with this PR its highlighting that we are inconsistent in our usage of `kekId`, `kekRef` and `kek` and we should standardise on one.